src/charm.py

```diff
@@ -519,7 +519,7 @@ def _reconcile_openstack_runners(self, runner_scaler: RunnerScaler) -> None:
     def _install_deps(self) -> None:
         """Install dependences for the charm."""
         logger.info("Installing charm dependencies.")
-        self._apt_install(["run-one", "python3-pip"])
+        self._apt_install(["run-one", "python3-pip", "python3-venv"])
```

Comment:
The venv package is needed for pipx.
Pipx can be installed, but not used without venv. Since pipx uses venv for virtual env for each executable python package.